pkg/kvstore/allocator/doublewrite/backend.go

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package doublewrite
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/allocator"
+	"github.com/cilium/cilium/pkg/idpool"
+	"github.com/cilium/cilium/pkg/k8s/identitybackend"
+	"github.com/cilium/cilium/pkg/kvstore"
+	kvstoreallocator "github.com/cilium/cilium/pkg/kvstore/allocator"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/rate"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "double-write-allocator")
+)
+
+func NewDoubleWriteBackend(c DoubleWriteBackendConfiguration) (allocator.Backend, error) {
```

Comment:
nit: Could you add a docblock here that gives a quick overview of what this does? Doesn't have to be fancy, just a few sentences would do. Thanks!

Reply:
Done!